hypothesis-python/tests/numpy/test_gufunc.py

```diff
@@ -0,0 +1,675 @@
+    signature = unparse((i_parsed_sig, o_parsed_sig))
+    # This is a 'private' function of np, so need to test it still works as we
+    # think it does.
+    inp, out = gu.parse_gufunc_signature(signature)
+
+    assert i_parsed_sig == inp
+    assert o_parsed_sig == out
+
+
+def test_check_set_like():
+    """Need for 100% coverage"""
+    assertInvalidArgument(gu._check_set_like, 0)
+    assertInvalidArgument(gu._check_set_like, "0")
+    assertInvalidArgument(gu._check_set_like, "foobar")
+
+
+@given(dictionaries(from_regex(VALID_DIM_NAMES), integers()), integers(), integers())
+def test_ddict_int_or_dict(D, default_val, default_val2):
+    DD = defaultdict(lambda: default_val, D)
+
+    DD2 = gu._int_or_dict(DD, default_val2)
+
+    # just pass thru
+    assert DD is DD2
+    # default_val2 is ignored
+    assert DD2.default_factory() == default_val
+
+
+@given(dictionaries(from_regex(VALID_DIM_NAMES), integers()), integers())
+def test_dict_int_or_dict(D, default_val):
+    DD = gu._int_or_dict(D, default_val)
+
+    assert DD == D
+    assert DD["---"] == default_val
+
+
+@given(integers(), integers())
+def test_int_int_or_dict(default_val, default_val2):
+    DD = gu._int_or_dict(default_val, default_val2)
+
+    assert len(DD) == 0
+    assert DD["---"] == default_val
+
+
+@given(real_scalar_dtypes(), _st_shape, data())
+def test_arrays(dtype, shape, data):
+    # unique argument to arrays gets tested in the tuple of arrays tests
+    choices = data.draw(real_from_dtype(dtype))
+
+    elements = sampled_from(choices)
+    S = gu._arrays(dtype, shape, elements)
+    X = data.draw(S)
+
+    assert np.shape(X) == shape
+    validate_elements([X], dtype=dtype, choices=choices)
+
+    assert type(X) == np.ndarray
+
+
+@given(real_scalar_dtypes(), _st_shape, data())
+def test_just_arrays(dtype, shape, data):
+    # unique argument to arrays gets tested in the tuple of arrays tests
+    choices = data.draw(real_from_dtype(dtype))
+
+    # test again, but this time pass in strategy to make sure it can handle it
+    elements = sampled_from(choices)
+    S = gu._arrays(just(dtype), just(shape), elements)
+    X = data.draw(S)
+
+    assert np.shape(X) == shape
+    validate_elements([X], dtype=dtype, choices=choices)
+
+    assert type(X) == np.ndarray
+
+
+# hypothesis.extra.numpy.array_shapes does not support 0 min_size so we roll
+# our own in this case.
+@given(
+    lists(_st_shape, min_size=0, max_size=5), real_scalar_dtypes(), booleans(), data()
+)
+def test_shapes_tuple_of_arrays(shapes, dtype, unique, data):
+    elements = from_dtype(np.dtype(dtype))
+
+    S = gu._tuple_of_arrays(shapes, dtype, elements=elements, unique=unique)
+    X = data.draw(S)
+
+    validate_elements(X, dtype=dtype, unique=unique)
+
+    assert len(shapes) == len(X)
+    for spec, drawn in zip(shapes, X):
+        assert tuple(spec) == np.shape(drawn)
+
+
+# hypothesis.extra.numpy.array_shapes does not support 0 min_size so we roll
```

Comment:
I think this is just another unforeseen use-case, so you could bump the floor size and dimensionality down to zero in `array_shapes`.  Doing that as *yet another* minor release is optional, as it's trivial, but would be nice.

Reply:
Are you thinking of changing the `order_check` or also the default arguments?